### PR TITLE
Feature/registration form

### PR DESCRIPTION
**Pull request by marcioszka**

### CODE REVIEW

File: src/components/registrationForm/RegistrationForm.css
Comment (Zachiel) [inline src/components/registrationForm/RegistrationForm.css:1]: Let's stick to one naming convention of classes

https://en.bem.info/methodology/naming-convention/

File: src/components/registrationForm/RegistrationForm.tsx [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+								getValues('confirmedPassword')
+						) {
+							setConfirmedPasswordValid(true);
+						}
+					}}
+				/>
+				{errors.confirmedPassword && (
+					<p className='validationPrompt'>Passwords must match.</p>
+				)}
+			</div>
```
Comment (Zachiel) [inline src/components/registrationForm/RegistrationForm.tsx]: Check Zod schema validation, saves a lot of code writing and improves readability :)

https://zod.dev/?id=strings
https://www.youtube.com/watch?v=dldjCPa9ZW4

File: src/components/registrationForm/RegistrationForm.tsx [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+		const passwordValue = getValues('password');
+
+		if (confirmedPasswordValue !== passwordValue) {
+			setError('confirmedPassword', { message: 'Passwords must match.' });
+			setConfirmedPasswordValid(false);
+		} else {
+			clearErrors('confirmedPassword');
+			setConfirmedPasswordValid(true);
+		}
+	};
```
Comment (Zachiel) [inline src/components/registrationForm/RegistrationForm.tsx]: This validation function can be extracted into a separate file where you pass states as porps

File: src/components/registrationForm/RegistrationForm.tsx [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+			// 	}
+			// } else {
+			// 	//console.log(response.status);
+			// 	alert('Thank you for registration!');
+			// 	setServerError(null);
+			// }
+		} catch (err) {
+			//setServerError(`${err}`);
+		}
+	};
```
Comment (Zachiel) [inline src/components/registrationForm/RegistrationForm.tsx]: Can be extracted to separate files:
Fetch (and any get, post, update, delete) into `@utils/fetchApi.ts` with props like url, states and setters,
Error handling into `@utils/apiErrorHandling.ts` passing in response code as prop,
finally submitForm imported `import { useSubmitForm } from '@hooks/useSubmitForm.ts' ` accepting required props that will be passed into utility functions

File: src/components/registrationForm/RegistrationForm.tsx (hunk 2 above)
Comment (marcioszka) [reply src/components/registrationForm/RegistrationForm.tsx]: Unnecessary while using zod library

**Review by Zachiel — COMMENTED** [review]
Looks good to me 😄 
I already reviewed the merge conflicts so if you need help with them just hit me up 👍

**Review by AlexanderKrysiuk — APPROVED** [review]

**Pull request merged**